File: P4/php/PegawaiHarian.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/Pegawai.php';

class PegawaiHarian extends Pegawai
{
	public function __construct(
		string $nip,
		string $nama,
		float $gajiPerHari,
		private readonly int $jumlahHari,
	) {
		parent::__construct($nip, $nama, $gajiPerHari);

		if ($this->jumlahHari < 0) {
			throw new InvalidArgumentException('Jumlah hari tidak boleh negatif');
		}
	}

	public function hitungGaji(): float
	{
		return parent::hitungGaji() * $this->jumlahHari;
	}

	public function jenis(): string { return 'HARIAN'; }
}

File: P4/java/PegawaiHarian.java
public class PegawaiHarian extends Pegawai {

	private final int jumlahHari;

   
	public PegawaiHarian(String nip, String nama, double gajiPerHari, int jumlahHari) {
		super(nip, nama, gajiPerHari);
		if (jumlahHari < 0) {
			throw new IllegalArgumentException("Jumlah hari tidak boleh negatif");
		}
		this.jumlahHari = jumlahHari;
	}

	@Override
	public double hitungGaji() {
		return super.hitungGaji() * jumlahHari;
	}

	@Override
	public String jenis() { return "HARIAN"; }
}
